pkg/keyspace/tso_keyspace_group.go

```diff
@@ -221,7 +256,7 @@ func (m *GroupManager) watchServiceAddrs(ctx context.Context, revision int64) (i
 			for _, event := range wresp.Events {
 				s := &discovery.ServiceRegistryEntry{}
 				if err := json.Unmarshal(event.Kv.Value, s); err != nil {
```

Comment:
Can't unmarshal here. see https://github.com/tikv/pd/pull/6346 Fixed bugs in tso service registry watching loop

Reply:
And use "revision = wresp.Header.Revision + 1" somewhere too.